### PR TITLE
Added more tests in test_models and test_routes

### PR DESCRIPTION
**Pull request by nijya**
1. Added more tests in `test_routes.py` to cover all corresponding route tests. 
2. Added two tests in `test_models.py` to cover serialize item test. 
Screen Shot 2023-07-03 at 17 47 31

### CODE REVIEW

File: tests/test_routes.py [hunk 1]
```diff
@@ -147,3 +149,106 @@ def test_list_shopcart_items_none_items_error(self, mock_shopcart_get_by_id):
...
+        self.assertEqual(res.status_code, status.HTTP_500_INTERNAL_SERVER_ERROR)
+    
+    @patch.object(Shopcart, 'update')
+    def test_add_shopcart_items_data_validation_error(self, mock_shopcart_update):
+        """It should return a 400 Bad Request response for DataValidationError"""
+        shopcart = self._create_an_empty_shopcart(1)[0]
+        item = ItemFactory()
+
+        # Raise a DataValidationError when calling shopcart.update
+        mock_shopcart_update.side_effect = DataValidationError("Invalid data")
```
Comment (wl2484) [inline tests/test_routes.py:246]: nice to learn there's such syntax!

Comment (wl2484) [reply tests/test_routes.py:246]: One reminder is that when selecting type of error to be raised as side effect, we would prefer choosing one that is possible to be triggered with the statement. For `shopcart.update()`, given that it is a method implemented in the SQLAlchemy package as one of the DBAPI, we should raise a possible error type according to their source code.

I checked SQLAlchemy's source code in Github and found that [SQLAlchemyError](https://github.com/sqlalchemy/sqlalchemy/blob/main/lib/sqlalchemy/exc.py#L10) is their base exception class lol

Comment (wl2484) [reply tests/test_routes.py:246]: I think the correct way here is to import the possible error class defined in SQLAlchemyError (e.g. the DBAPIError), and set it as the side_effect. However, I haven't succeeded in figuring out how to import the DBAPIError class, that's why I chose Exception("DBAPIError") as an workaround in other test cases 😂

**Review by wl2484 — COMMENTED** [review]
The test cases are awesome, thank you @nijya Zihan!
I just added a couple of comments to some minor issues.

File: tests/test_routes.py (hunk 1 above)
Comment (nijya) [reply tests/test_routes.py:246]: Thanks for your comments!

Comment (wl2484) [reply tests/test_routes.py:246]: ok, maybe let's leave it as is until we figure out a better way, thanks~

**Review by wl2484 — APPROVED** [review]

**Pull request merged**